src/spec/create_extension_spec.py

```diff
@@ -1,47 +1,394 @@
+                                                                              'and Layers hold different stimulus waveform types. '
+                                                                              'See also https://alleninstitute.github.io/MIES/file/_m_i_e_s___wave_data_folder_getters_8ipf.html#_CPPv423GetWaveBuilderWaveParamv.',
+                                                                              neurodata_type_inc='StimulusSetWavebuilderParameter'
+                                                                              ),
+                                                               NWBDatasetSpec(doc='Stimulus set waveform data. This is only present if not all three parameter waves could be found '
+                                                                              'or a third-party stimset was used. One column per sweep.',
+                                                                              neurodata_type_inc='StimulusSetWaveform'
+                                                                              ),
+                                                               ]
+                                                     )
+                                        ],
+                                )
+
     # these arguments were auto-generated from your cookiecutter inputs
     ns_builder = NWBNamespaceBuilder(
-        doc="""An NWB:N extension for data and metadata from the Multichannel Igor Electrophysiology Suite (MIES)""",
-        name="""ndx-MIES""",
+        doc=("""An NWB:N extension for data and metadata from the Multichannel
+            Igor Electrophysiology Suite (MIES)"""),
+        name="""ndx-mies""",
         version="""0.1.0""",
         author=list(map(str.strip, """Thomas Braun""".split(','))),
         contact=list(map(str.strip, """thomas.braun@byte-physics.de""".split(',')))
     )
 
-    # TODO: specify the neurodata_types that are used by the extension as well
-    # as in which namespace they are found
-    # this is similar to specifying the Python modules that need to be imported
-    # to use your new data types
-    ns_builder.include_type('ElectricalSeries', namespace='core')
-
-    # TODO: define your new data types
-    # see https://pynwb.readthedocs.io/en/latest/extensions.html#extending-nwb
-    # for more information
-    tetrode_series = NWBGroupSpec(
-        neurodata_type_def='TetrodeSeries',
-        neurodata_type_inc='ElectricalSeries',
-        doc=('An extension of ElectricalSeries to include the tetrode ID for '
-             'each time series.'),
-        attributes=[
-            NWBAttributeSpec(
-                name='trode_id',
-                doc='The tetrode ID.',
-                dtype='int32'
-            )
-        ],
-    )
+    # include referenced types
+    ns_builder.include_type('Data', namespace='hdmf-common')
```

Comment:
Are all of these types really used/referenced? I don't see many of them referenced

Reply:
I could not get it working with fewer the last time I tried.